packages/ui/src/components/va-tree-view/hooks/useTreeBuilder.ts

```diff
@@ -0,0 +1,37 @@
+import { TreeBuilderFunc, CreateNodeFunc, UseTreeBuilderFunc } from '../types'
+import { reactive } from 'vue'
+
+export const createNode: CreateNodeFunc = ({
+  node,
+  children = [],
+  expandAll = false,
+  level = 0,
+}) => ({
+  ...node,
+  children,
+  level,
+  hasChildren: !!children.length,
```

Comment:
```ts
key: getKey(node, props.keyBy) || fallbackKey,
label: getLabel(node, props.labelBy) || fallbackLabel,
```
where `fallbackKey` and `fallbackLabel` may include the key and label of parentNode, respectively, and the index of the current node in the node array.